### PR TITLE
feat(cluster): Extend the available attributes for the `proxmox_virtual_environment_cluster_options` resource

### PR DESCRIPTION
**Pull request by svengreb**
### Contributor's Note

- [x] I have added / updated documentation in `/docs` for any user-facing features or additions.
- [x] I have added / updated acceptance tests in `/fwprovider/tests` for any new or updated resources / data sources.
- [x] I have ran `make example` to verify that the change works as expected.

### Summary

This pull request implements the `next-id` and `notify` [PVE API cluster options](https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/options).

The `next-id` attribute allows to control the range for the next free VM ID. It is implemented as object and can be used in the `proxmox_virtual_environment_cluster_options` resource like this:

```terraform
resource "proxmox_virtual_environment_cluster_options" "options" {
  next_id = {
    lower = 200
    upper = 299
  }
}
```

Note that **the minimum and maximum values are unfortunately not documented in the PVE API explorer** but can be found in the web UI where the form fields have validations!

The `notify` PVE API attribute is also an object that has all the PVE API fields:

```terraform
resource "proxmox_virtual_environment_cluster_options" "options" {
  notify = {
    ha_fencing_mode            = "never"
    ha_fencing_target          = "default-matcher"
    package_updates            = "always"
    package_updates_target     = "default-matcher"
    package_replication        = "always"
    package_replication_target = "default-matcher"
  }
}
```

Note that the "fencing" attribute names have been adjusted to better reflect their meaning since they are scoped to [the Proxmox VE HA fencing feature](https://pve.proxmox.com/wiki/Fencing). All attributes with the `_target` suffix are names for the [Proxmox VE notifications matchers](https://pve.proxmox.com/pve-docs/chapter-notifications.html#notification_matchers).

### Community Note

- Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by bpg** [comment]
Thanks for the PR! 👍🏼 I'll review it over the weekend.

Sorry about the failed Qodana checks, I'm still experimenting with different static code analysis tools.

> Note that the minimum and maximum values are unfortunately not documented in the PVE API explorer but can be found in the web UI where the form fields have validations!

Just fyi, it is documented in https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/qemu (POST), and also in many references in the admin guide https://pve.proxmox.com/pve-docs/pve-admin-guide.html#_strong_qm_strong_qemu_kvm_virtual_machine_manager

**Comment by bpg** [comment]
@all-contributors please add @svengreb for code

**Comment by svengreb** [comment]
Sure, take your time 😃

> Just fyi, it is documented in [https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/qemu](https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/%7Bnode%7D/qemu) (POST), and also in many references in the admin guide https://pve.proxmox.com/pve-docs/pve-admin-guide.html#_strong_qm_strong_qemu_kvm_virtual_machine_manager

Oh, I haven't checked other endpoints, but I makes sense that these numbers must be documented somewhere related to VMs itself!
I'll update the notes I added to the code regarding these numbers.

File: fwprovider/resource_options.go [hunk 1]
```diff
@@ -414,6 +550,108 @@ func (r *clusterOptionsResource) Schema(
...
+					},
+					"ha_fencing_target": schema.StringAttribute{
+						Description:         "Cluster-wide notification settings for the HA fencing target.",
+						MarkdownDescription: "Cluster-wide notification settings for the HA fencing target.",
+						Optional:            true,
+						Validators: []validator.String{
+							stringvalidator.UTF8LengthAtLeast(1),
+							stringvalidator.RegexMatches(regexp.MustCompile(`^\S|^$`), "must not start with whitespace"),
+							stringvalidator.RegexMatches(regexp.MustCompile(`\S$|^$`), "must not end with whitespace"),
+						},
```
Comment (bpg-dev) [inline fwprovider/resource_options.go:600]: This validation is duplicated few times in the schema, so could extracted in a shared validator.

File: fwprovider/resource_options.go [hunk 2]
```diff
@@ -285,6 +403,24 @@ func (m *clusterOptionsModel) importFromOptionsAPI(
 		m.MigrationNetwork = types.StringNull()
 	}
 
+	if opts.NextID != nil {
+		m.NextID = &clusterOptionsNextIDModel{}
+		lower := int64(*opts.NextID.Lower)
```
Comment (bpg-dev) [inline fwprovider/resource_options.go:408]: we can add something like `.PointerInt64()` to `CustomInt64` type to simplify this conversion.

**Review by bpg — APPROVED** [review]
Thanks so much for adding this feature!

There are few minor comments that we can address later. I'll probably incorporate them into my ongoing work on the VM resource.

LGTM! 🚀

File: fwprovider/resource_options.go (hunk 1 above)
Comment (svengreb) [reply fwprovider/resource_options.go:600]: Definitely, it's basically a validator that could be applied to (almost) all strings, even when the Proxmox VE API does not check or enforce it.

File: fwprovider/resource_options.go (hunk 2 above)
Comment (svengreb) [reply fwprovider/resource_options.go:408]: Yep, I also thought about it, but decided to solve it this way to get the implementation done.

**Pull request merged**